### PR TITLE
Prevent torch_learner.py from crashing under some parameter-freezing edge cases.

### PR DESCRIPTION
**Pull request by MatthewCWeston**
## Description

When fine-tuning pretrained models, some parameters might be frozen, while others aren't. In some cases, under some algorithms, this may result in training runs where no parameters with gradients are used in a given iteration.

This fix prevents this crash by avoiding calls to `backward()` when no gradient exists on total_loss.

### CODE REVIEW

File: rllib/core/learner/torch/torch_learner.py [hunk 1]
```diff
@@ -187,6 +187,10 @@ def compute_gradients(
             assert total_loss == 0
             return {}
 
+        # If all parameters for this module are currently frozen, we can't optimize it.
```
[the remark this answers is not in the document]
Comment (MatthewCWeston) [reply rllib/core/learner/torch/torch_learner.py:190]: I don't think that's an improvement, but the maintainers are welcome to add it if desired.

**Review by pseudo-rnd-thoughts — APPROVED** [review]
I'm surprised that there is a problem for this and we should skip the training of the policy completely.
Happy to merge this @MatthewCWeston but could you give some details on when this is happening

**Comment by MatthewCWeston** [comment]
> I'm surprised that there is a problem for this and we should skip the training of the policy completely. Happy to merge this @MatthewCWeston but could you give some details on when this is happening

It's a bit a weird edge-case. I'm doing some weight surgery _(a la [OpenAI Five](https://arxiv.org/abs/1912.06680))_, and this involves freezing pretrained model weights while adding new ones, some of which, depending on batch size and the surgery being performed, might not show up in every training iteration when working with certain odd model architectures that don't necessarily use all of their parameters in a given pass _(e.g. a transformer encoder that's learning the embedding of a new token)._
